mrmustard/lab/representations/converter.py

```diff
@@ -0,0 +1,497 @@
+        and `B` has shape `2M`).
+        The order of the rows/columns of A and B corresponds to a density matrix with the usual
+        ordering of the indices.
+
+        Note that here A and B are defined with respect to the literature.
+
+        Args:
+            wigner_dm (WignerDM) : the Wigner DM representation of the state
+
+        Returns:
+            The BargmannDM representation of the input state
+
+        """
+        N = wigner_dm.data.cov.shape[-1] // 2
+
+        A = math.matmul(math.Xmat(N), self._cayley(self._pq_to_aadag(wigner_dm.data.cov), c=0.5))
+        Q, beta = self._wigner_to_husimi(wigner_dm.data.cov, wigner_dm.data.means)
+        b = math.solve(Q, beta)
+        B = math.conj(b)
+        C = math.exp(-0.5 * math.sum(math.conj(beta) * b)) / math.sqrt(math.det(Q))
+
+        return BargmannDM(A=A, b=B, c=C)
+
+    def _wignerket_to_bargmannket(self, wigner_ket: WignerKet) -> BargmannKet:
+        r"""
+        Converts the Wigner representation in terms of covariance matrix and mean vector into the
+        Bargmann A,B,C triple for a Hilbert vector (i.e. for M modes, A has shape M x M and B has
+        shape M).
+
+        Args:
+            wigner_ket (WignerKet) : the input Wigner representation of the state
+
+        Returns:
+            The Bargmann representation of the input state
+        """
+
+        N = wigner_ket.data.symplectic.shape[-1] // 2
+        cov = wigner_ket.cov
+        means = wigner_ket.means
+        wigner_dm = WignerDM(cov=cov, means=means)
+        bargmann_dm = self._wignerdm_to_bargmanndm(wigner_dm=wigner_dm)
+
+        return BargmannKet(
+            bargmann_dm.data.A[N:, N:], bargmann_dm.data.b[N:], math.sqrt(bargmann_dm.data.c)
+        )
+
+    ########################################################################
+    ###                     From Wigner to Fock                         ###
+    ########################################################################
+
+    def _wignerket_to_fockket(
+        self,
+        wignerket: WignerKet,
+        max_prob: float = 1.0,
+        max_photon: Optional[int] = None,
+        cutoffs: List[int] = None,
+    ) -> FockKet:
+        r"""
+        Returns the Fock representation of a Gaussian state in ket form.
+
+        Args:
+            wignerket (WignerKet)       : the WignerKet object.
+            cutoffs (List[int]) .       : the shape of the desired Fock tensor
+            max_prob (float)            : the maximum probability of a the state (applies only if
+                                          the ket is returned)
+            max_photon (Optional[int]) : the maximum number of photons in the state (applies only
+                                          if the ket is returned)
+
+        Returns:
+            FockKet: the fock representation of the ket.
+        """
+
+        bargmann_ket = self._wignerket_to_bargmannket(wignerket)
```

Comment:
why not define Bargmann -> Fock on its own, and then use it to go from Wigner to Fock?

Reply:
Good point. I will do it.

Reply:
Now the Bargmann -> X paths are very important because Bargmann is going to become default